src/mpi/mpi_caf.c

```diff
@@ -912,37 +912,42 @@ PREFIX (sendget) (caf_token_t token_s, size_t offset_s, int image_index_s,
           ptrdiff_t array_offset_dst = 0;
           ptrdiff_t stride = 1;
           ptrdiff_t extent = 1;
+	  ptrdiff_t tot_ext = 1;
           for (j = 0; j < rank-1; j++)
             {
-              array_offset_dst += ((i / (extent*stride))
+              array_offset_dst += ((i / tot_ext)
                                    % (dest->dim[j]._ubound
                                       - dest->dim[j].lower_bound + 1))
                 * dest->dim[j]._stride;
               extent = (dest->dim[j]._ubound - dest->dim[j].lower_bound + 1);
               stride = dest->dim[j]._stride;
+	      tot_ext *= extent;
             }
 
+	  array_offset_dst += (i / tot_ext) * dest->dim[rank-1]._stride;
+	  
 	  extent = (dest->dim[rank-1]._ubound - dest->dim[rank-1].lower_bound + 1);
```

Comment:
Oh, sorry, I haven't used githubs review caps yet. So my comment went to the wrong line. Not the line starting with `array_offset_dst +=` should be removed, but the line starting with `extent = `. Extent is not read after set here anymore.